packages/devtools_app/lib/src/ui/flutter/flutter_icon_renderer.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Library to render [DevToolsIcon]s as [Widget]s.
+library icon_renderer;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import '../icons.dart';
+import '../material_icons.dart';
+import '../theme.dart';
+
+final Expando<Widget> widgetExpando = Expando('IconRenderer');
+
+typedef DrawIconImageCallback = void Function(Canvas canvas);
+
+String _rewriteIconAssetPath(String path) {
+  assert(path.startsWith('/'));
+  // Paths under flutter will include the web directory while paths for the
+  // legacy dart:html app do not.
+  path = 'web$path';
+  if (path.endsWith('.png') && !path.endsWith('@2x.png')) {
+    // By convention icons all have high DPI verisons with @2x added to the
+    // file name.
+    // We could opt out of the @2x versions for low DPI devices but there isn't
+    // really much harm in using the higher resolution icons on lower DPI
+    // displays.
+    return '${path.substring(0, path.length - 4)}@2x.png';
+  }
+  return path;
+}
+
+class _ColorIconPainter extends CustomPainter {
+  const _ColorIconPainter(this.icon);
+
+  final ColorIcon icon;
+
+  static const double iconMargin = 1;
+
+  Color get color => icon.color;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    // draw a black and gray grid to use as the background to disambiguate
+    // opaque colors from translucent colors.
+    final greyPaint = Paint()..color = grey;
+    final iconRect = Rect.fromLTRB(
+      iconMargin,
+      iconMargin,
+      size.width - iconMargin,
+      size.height - iconMargin,
```

Comment:
Is subtracting `iconMargin` correct here? That would make the width of the rect `size.width - 2*iconMargin`?  Adding `iconMargin` would keep the width of the rect as `size.width`.

Reply:
The math works because `Rect.fromLTRB` is used rather than LTWH.

Reply:
With this calculation, the icon rect will be `2*iconMargin` smaller in both height and width than what is specified by `size`. As long as that is intentional and not a side effect, lgtm.

Reply:
this probably should be called padding instead of margin.
The idea is the icon's size is 
size.width, size.height
and the icon iself is draw in the center of that area
with iconMargin whitespace between the icon and the size.